static/events.ts

```diff
@@ -0,0 +1,151 @@
+    editorClose: (editorId: number) => void;
+    editorLinkLine: (editorId: number, lineNumber: number, colBegin: number, colEnd: number, reveal: boolean) => void;
+    editorOpen: (editorId: number) => void;
+    editorSetDecoration: (editorId: number, lineNumber: number, reveal: boolean) => void;
+    executeResult: (executorId: number, compiler: unknown, result: unknown, languages?: Language) => void;
+    executor: (
+        executorId: number,
+        compiler: unknown,
+        options: string,
+        editorId: boolean | number,
+        treeId: boolean | number
+    ) => void;
+    executorClose: (executorId: number) => void;
+    executorOpen: (executorId: number, editorId: boolean | number) => void;
+    filtersChange: (compilerId: number, filters: CompilerFilters) => void;
+    findCompilers: () => void;
+    findEditors: () => void;
+    findExecutors: () => void;
+    flagsViewClosed: (compilerId: number, options: string) => void;
+    flagsViewOpened: (compilerId: number) => void;
+    gccDumpFiltersChanged: (compilerId: number, filters: CompilerFilters, recompile: boolean) => void;
+    // TODO: pass type
+    gccDumpPassSelected: (compilerId: number, pass: string, recompile: boolean) => void;
+    gccDumpUIInit: (compilerId: number) => void;
+    gccDumpViewClosed: (compilerId: number) => void;
+    gccDumpViewOpened: (compilerId: number) => void;
+    gnatDebugTreeViewClosed: (compilerId: number) => void;
+    gnatDebugTreeViewOpened: (compilerId: number) => void;
+    gnatDebugViewClosed: (compilerId: number) => void;
+    gnatDebugViewOpened: (compilerId: number) => void;
+    haskellCmmViewClosed: (compilerId: number) => void;
+    haskellCmmViewOpened: (compilerId: number) => void;
+    haskellCoreViewClosed: (compilerId: number) => void;
+    haskellCoreViewOpened: (compilerId: number) => void;
+    haskellStgViewClosed: (compilerId: number) => void;
+    haskellStgViewOpened: (compilerId: number) => void;
+    initialised: () => void;
+    irViewClosed: (compilerId: number) => void;
+    irViewOpened: (compilerId: number) => void;
+    languageChange: (editorId: number, newLangId: string, treeId?: boolean | number) => void;
+    modifySettings: (modifiedSettings: Partial<SiteSettings>) => void;
+    // TODO: Add type specificity
+    motd: (data: unknown) => void;
+    newSource: (editorId: number, newSource: string) => void;
+    optViewClosed: (compilerId: number) => void;
+    optViewOpened: (compilerId: number) => void;
+    outputClosed: (compilerId: number) => void;
+    outputOpened: (compilerId: number) => void;
+    panesLinkLine: (
+        compilerId: number,
+        lineNumber: number,
+        colBegin: number,
+        colEnd: number,
+        reveal: boolean,
+        sender: string
+    ) => void;
+    ppViewClosed: (compilerId: number) => void;
+    ppViewOpened: (compilerId: number) => void;
+    // TODO: See if this is necessary. Nothing emits this event
```

Comment:
I do believe this is emitted: https://github.com/compiler-explorer/compiler-explorer/blob/335f683b6b065072061496dd63e31201bece108d/static/panes/pp-view.ts#L88-L96